### ISSUE TITLE
[0.10.8] Indentation adds a tab on the beginning

### ISSUE THREAD

**Issue by philipgiuliani** [Microsoft/vscode#2845]
Steps to reproduce 100%:
1. Enable `"files.trimTrailingWhitespace": true`
2. Create a `.js` file
3. Indent somewhere in a function, move the cursor to another line and save the file so that the whitespace gets trimmed.
4. Go back to that line and press TAB.  indentation

Here two examples of what is happening:

![indentation_v2](https://cloud.githubusercontent.com/assets/5446019/12918585/2b1e1b40-cf40-11e5-941e-bc3bc223e57f.gif)
![indentation](https://cloud.githubusercontent.com/assets/5446019/12918584/2b1a5794-cf40-11e5-9610-3041b3bed789.gif)

**Comment by SamVerschueren** [comment]
//cc @bpasero @alexandrudima

**Comment by nchammas** [comment]
I'm seeing a potentially related issue with SQL files if I highlight some lines and try to indent them with tab.

![screen recording 2016-02-09 at 03 27 pm](https://cloud.githubusercontent.com/assets/1039369/12929612/b2f7de88-cf41-11e5-9280-9560a9be490e.gif)


This one is strange because, in addition to killing all the text, Command + Z refuses to undo the action!

**Comment by philipgiuliani** [comment]
@nchammas Thats the following issue i think https://github.com/Microsoft/vscode/issues/2829

**Comment by nchammas** [comment]
Oh, good catch!

**Comment by mbroadst** [comment]
+1, seeing this as well

**Comment by alexandrudima** [comment]
@mbroadst What OS are you on? I am trying to reproduce in #2829, so far unsuccessfully. Do you have other steps?

**Comment by SamVerschueren** [comment]
@alexandrudima This bug seems only related in a file with `insertSpaces` set to `true`. In a file with tabs, this isn't an issue. Just so you know :).

**Comment by mbroadst** [comment]
@alexandrudima I'm on OSX, vscode version 0.10.8. I just tried to make a gif screen grab of the behavior, and couldn't reproduce it every time though. It seems almost random, and during my attempts I also ran into some weird indentation bug where after I saved the file, and then indented, it indented in the middle of the string randomly.

my user settings are:
```
{
  "javascript.validate.enable": false,
  "jshint.enable": true,
  "editor.renderWhitespace": true,
  "editor.insertSpaces": true,
  "editor.tabSize": 2,
  "files.trimTrailingWhitespace": true
}
```
